src/object/srv_obj_migrate.c

```diff
@@ -856,7 +856,7 @@ migrate_fetch_update_parity(struct migrate_one *mrone, daos_handle_t oh,
 	char		*data;
 	daos_size_t	 size;
 	unsigned int	 p = obj_ec_parity_tgt_nr(oca);
-	unsigned char	*p_bufs[p];
```

Comment:
oh, I just realized that variable length array is supported since C99, so what's the exact bug it caused?

Reply:
I am not very sure either, I guess that "variable length array is supported since C99" is only valid when pass the length as an input parameter of the func, right?  L858 need run-time calculation after enter the func.
here I met segfault in real test (for manually added EC 4+3 test) and no that issue after this small PR.

Reply:
In C99, the length parameter must come before the variable-length array parameter in function calls.
https://en.wikipedia.org/wiki/Variable-length_array